src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -778,6 +778,27 @@ public void CompileMethod(MethodWithGCInfo methodCodeNodeNeedingCode, Logger log
                     return;
                 }
 
+                if (MethodBeingCompiled.IsAsyncThunk())
```

Comment:
If the async thunks are inlined, I believe we can encounter references for the managed helpers in methods that are not async thunks. We should not be limiting this to async thunks.